File: benchmarks/haskell/app/GroupBy.hs
module Main where

import DataProcessing.GroupBy (groupByCategory)
import Control.DeepSeq (rnf)
import Control.Exception (evaluate)
import Options.Applicative

newtype Args = Args { argInput :: FilePath }

argsParser :: ParserInfo Args
argsParser = info (Args <$> strOption (long "input" <> metavar "FILE") <**> helper) mempty

main :: IO ()
main = do
    Args path <- execParser argsParser
    totals <- groupByCategory path
    _ <- evaluate (rnf totals)
    putStrLn $ show (length totals) ++ " categories"

File: benchmarks/rust/src/bin/group_by.rs
use clap::Parser;
use fp_benchmarks::data_processing::group_by;
use std::path::PathBuf;

#[derive(Parser)]
struct Args {
    #[arg(long)]
    input: PathBuf,
}

fn main() {
    let args = Args::parse();
    let totals = group_by::group_by_category(&args.input);
    println!("{} categories", totals.len());
}
